scripts/start.js

```diff
@@ -66,62 +69,96 @@ function clearConsole() {
 }
 
 var compiler = webpack(config, handleCompile);
-compiler.plugin('invalid', function () {
-  clearConsole();
-  console.log('Compiling...');
-});
-compiler.plugin('done', function (stats) {
-  clearConsole();
-  var hasErrors = stats.hasErrors();
-  var hasWarnings = stats.hasWarnings();
-  if (!hasErrors && !hasWarnings) {
-    console.log(chalk.green('Compiled successfully!'));
-    console.log();
-    console.log('The app is running at http://localhost:3000/');
-    console.log();
-    return;
-  }
-
-  var json = stats.toJson();
-  var formattedErrors = json.errors.map(message =>
-    'Error in ' + formatMessage(message)
-  );
-  var formattedWarnings = json.warnings.map(message =>
-    'Warning in ' + formatMessage(message)
-  );
 
-  if (hasErrors) {
-    console.log(chalk.red('Failed to compile.'));
-    console.log();
-    if (formattedErrors.some(isLikelyASyntaxError)) {
-      // If there are any syntax errors, show just them.
-      // This prevents a confusing ESLint parsing error
-      // preceding a much more useful Babel syntax error.
-      formattedErrors = formattedErrors.filter(isLikelyASyntaxError);
+function promptForPort(suggestedPort) {
+  return new Promise((resolve, reject) => {
+    if (DEFAULT_PORT !== suggestedPort) {
+      var rl = readline.createInterface({
+        input: process.stdin,
+        output: process.stdout
+      });
+
+      var question = chalk.red('Something is already running at port ' + suggestedPort) +
+        '\nWould you like to run the app at another port instead? [Y/n] ';
+
+      rl.question(question, answer => {
+        var shouldChangePort = (
+          answer.length === 0 ||
+          answer.match(/yes|y/i)
+        );
+
+        if (shouldChangePort) {
+          // Replace the port in webpack config
+          config.entry.map(c => c.replace(/(\d+)/g, suggestedPort));
```

Comment:
I would prefer we do this without mutation because this can be confusing later on. Alternative: shallowly clone the config in setupCompiler. And move compiler creation there.